### DISCUSSION TITLE
Upgrade from 7.1.2(tasmota)  and flash size confusion

### DISCUSSION THREAD

**Discussion by Moosbueffel** [arendst/Tasmota#11841] [Q&A] [answered]
I'm not sure, what to do. I have to upgrade from V7.1.2 to the current version.

If I understand the doc, I cannot use OTA update, because this is than always the last/current version?
There are some inconsistencies:
1) In the upgrade description I can read that I can't use gziped files and:
- that I **have to** follow the upgrade path
   [here ](https://tasmota.github.io/docs/Upgrading/)I have to upgrade first to V8,5,1 ??
   [here ](https://github.com/arendst/Tasmota/releases/tag/v8.1.0)I have to upgrade first to V8.1 ??

I tried to upgrade with the downloaded .bin file V8.1.0. As expected, I got the error: "Upload buffer miscompare" ´. So I have to download and upgrade to tasmota-minimal.bin.

I only found [this ](https://ota.tasmota.com/tasmota/)minimal-version which is the current version V9.x. So I can upgrade to this and than directly to the full version V9.x? Or do I need also using the upgrade path to V8.1 - V8.5.1 - V9.1 - current version??

2 The second question is to the flash size. My Sonoff POW 2 show me this:
![grafik](https://user-images.githubusercontent.com/7726457/115732342-2f565980-a388-11eb-9051-b7f6270ca81c.png)

There are 4MB flash, but only 1MB usable?

**Comment by meingraham** [comment, marked as answer]
You can upgrade using OTA but you must use file upload rather than web upload. First you download the necessary binary files to your computer. Then you OTA update by selecting the binaries from your computer.

I am leery of skipping anything so I typically will upload tasmota-minimal.bin followed by tasmota.bin for each intermediate version and a `RESET 6` between EVERY file upload. Likely overkill... but better safe than sorry.

You cannot use the .gz binaries until you are at at least Tasmota 8.2 or higher.

**Reply by sfromis** [reply]
The easier upgrade strategy from such an old version is to switch the the smallest non-minimal binary on the old version (normally tasmota-lite), follow the recommended steps using smallest, and then upgrade to full Tasmota after getting to current version. This saves you the trouble of doing two upgrades for each step, while still migrating your settings. The `Reset 6` strategy should ideally not be needed, but of course there is even less which could go wrong that way. If serial flashing is feasible, that would be by first choice instead of the stepwise upgrading.

Yes, only 1MB "reliably usable" with Tasmota due to compatibility with 1MB devices. You may have flashed your device with a 4MB memory model which has a risk of providing "additional challenges" during upgrading, when you're not using the standard 1MB memory model. At least, the Sonoff Pow R2 is reasonably easy to do wired serial erase/flash on, if the need arises.

**Comment by Moosbueffel** [comment]
Many thanks, for your advice. It works :-). 

Serial flashing was difficult, because device is built in. I did upgrade to current version step by step with minimal version. And the last step to full version. (No "RESET 6"). And it works fine and fast.

**Reply by barbudor** [reply]
So you didn't followed sfromis recommendation to use _the smallest **NON**-minimal version_
Glad it worked, you've been lucky 

You should NEVER use tasmota-minimal for an upgrade but tasmota-lite as written by sfromis.

**Reply by meingraham** [reply]
@Moosbueffel

tasmota-minimal is an option **_ONLY_** if the device had Tasmota on it previously and the Wi-Fi parameters are retained (e.g., `RESET 6`). Otherwise you will end up with a Tasmota version that you cannot connect to wirelessly to provide settings. In your case, your upgrade allowed you to do that. Otherwise, tasmota-lite is the "safest intermediate" variant if there is flash space for it to upload successfully. If things had gone awry for you, the only way to recover would have been to flash via the serial interface (i.e., connect wires).

**Comment by Moosbueffel** [comment]
Oh - I haven't realized it. In my mind was the minimal version. I have to better read in future!
